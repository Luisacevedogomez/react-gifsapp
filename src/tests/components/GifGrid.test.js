import '@testing-library/jest-dom';
import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe('Pruebas componente <GifGrid/>', () => {

    const category = 'One Piece';
    test('debe cargar correctamente el componente', () => {

        useFetchGifs.mockReturnValue({
          data: [],
          loading: true,
        });

        const wrapper = shallow(<GifGrid category={category} />);

        expect(wrapper).toMatchSnapshot();

    })

    test('debe de mostrar items cuando se cargan las imagenes en useFetchGifs', () => {

        const gifs = [{
            id: 'ABC',
            title: 'Cualquier titulo',
            url: 'https://localhost/cualquier/cosa.jpg'
        }]

        useFetchGifs.mockReturnValue({
          data: gifs,
          loading: false,
        });
        
        const wrapper = shallow(<GifGrid category={category} />);

        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    })
    

})
