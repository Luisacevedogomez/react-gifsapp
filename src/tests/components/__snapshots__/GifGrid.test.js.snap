// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas componente <GifGrid/> debe cargar correctamente el componente 1`] = `
<Fragment>
  <h3
    className="card animate__animated animate__fadeIn"
  >
    One Piece
  </h3>
  <div
    className="card-grid"
  />
</Fragment>
`;
